karton/core/utils.py

```diff
@@ -12,6 +12,55 @@ def chunks(seq: Sequence[T], size: int) -> Iterator[Sequence[T]]:
     return (seq[pos : pos + size] for pos in range(0, len(seq), size))
 
 
+def recursive_iter(obj) -> Iterator[Any]:
+    """
+    Yields all values recursively from nested list/dict structures
+
+    :param obj: Object to iterate over
+    """
+    if isinstance(obj, (list, tuple, set)):
```

Comment:
Do we really have to care about tuples and sets since they aren't directly json-serializable?

Reply:
Tuples are serializable, but sets are not. But I added them for completion + they will be converted to lists that are serializable.

Reply:
After internal discussion: removed sets from supported "iterable" structures